datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -665,39 +871,19 @@ func buildCreateStandardPayloadQueries(d *schema.ResourceData) []datadogV2.Secur
 		}
 
 		if v, ok := query["group_by_fields"]; ok {
-			tfGroupByFields := v.([]interface{})
-			groupByFields := make([]string, 0)
-			for _, value := range tfGroupByFields {
-				if value != nil {
-					groupByFields = append(groupByFields, value.(string))
-				}
-			}
-			payloadQuery.SetGroupByFields(groupByFields)
+			payloadQuery.SetGroupByFields(parseStringArray(v.([]interface{})))
 		}
 
 		if v, ok := query["distinct_fields"]; ok {
-			tfDistinctFields := v.([]interface{})
-			distinctFields := make([]string, 0)
-			for _, value := range tfDistinctFields {
-				if value != nil {
-					distinctFields = append(distinctFields, value.(string))
-				}
-			}
-			payloadQuery.SetDistinctFields(distinctFields)
+			payloadQuery.SetDistinctFields(parseStringArray(v.([]interface{})))
 		}
 
 		if v, ok := query["metric"]; ok {
 			payloadQuery.SetMetric(v.(string))
 		}
 
 		if v, ok := query["metrics"]; ok && v != nil {
-			if tfMetrics, ok := v.([]interface{}); ok && len(tfMetrics) > 0 {
-				metrics := make([]string, len(tfMetrics))
-				for i, value := range tfMetrics {
-					metrics[i] = value.(string)
-				}
-				payloadQuery.SetMetrics(metrics)
-			}
+			payloadQuery.SetMetrics(parseStringArray(v.([]interface{})))
```

Comment:
This changes the behavior slightly, in the payload this used to omit metrics if the list was length 0, but now the metrics field gets explicitly set with `"metrics": [],`. Is this intentional?

Reply:
Good catch, it wasn't really intentional. However I checked the API, it treats empty arrays and omitted fields similarly, so let's keep it like this.